lms/djangoapps/course_wiki/middleware.py

```diff
@@ -75,6 +76,13 @@ def process_view(self, request, view_func, view_args, view_kwargs):  # pylint: d
                     # if a user is logged in, but not authorized to see a page,
                     # we'll redirect them to the course about page
                     return redirect('about_course', course_id.to_deprecated_string())
+
+                # If enterprise integration is enabled, and course_id requires data sharing consent,
```

Comment:
Folding the two methods together simplifies things nicely, but it does require changes in the other courseware views.  Please see 1687819.